### PR TITLE
feat: implement automated release management improvements

### PR DESCRIPTION
**Pull request by scott45**
# SPDX-License-Identifier: Apache-2.0

## What did we change?
Implement cicd working group changes

## Why are we doing this?
cicd recon

## How was it tested?
- [x] Locally
- [x] Development Environment
- [ ] Not needed, changes very basic
- [ ] Husky successfully run
- [ ] Unit tests passing and Documentation done

### CODE REVIEW

**Review by Sandy-at-Tazama — DISMISSED** [review]

File: .github/workflows/npm-publish-main.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        uses: actions/checkout@v4
+
+      # Setup .npmrc file to publish to GitHub Packages
+      - name: Setup Node.js (.npmrc)
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+          registry-url: 'https://npm.pkg.github.com'
+          # Defaults to the user or organization that owns the workflow file
+          scope: '@scott45'
```
Comment (rtkay123) [inline .github/workflows/npm-publish-main.yml]: Shouldn't this point to `tazama-lf`?

File: .github/workflows/npm-publish-dev.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Configure Git
+      - name: Git configuration
+        run: |
+          git config --global user.email "busingescott@gmail.com"
```
Comment (rtkay123) [inline .github/workflows/npm-publish-dev.yml]: I think we should use a bot here as well

**Review by rtkay123 — COMMENTED** [review]
I left some comments in the code

File: .github/workflows/npm-publish-main.yml (hunk 1 above)
Comment (scott45) [reply .github/workflows/npm-publish-main.yml]: Yes Rodney, PR still a draft. I've cleaned all that up. I'll be pushing the changes

File: .github/workflows/npm-publish-dev.yml (hunk 2 above)
Comment (scott45) [reply .github/workflows/npm-publish-dev.yml]: Yes Rod, PR still a draft. I've cleaned all that up. I'll be pushing the changes

**Review by Sandy-at-Tazama — DISMISSED** [review]

**Review by Sandy-at-Tazama — APPROVED** [review]

**Pull request merged**